src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -94,19 +94,24 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                var testExecution = RunTestOnSeparateThread(context);
-                if (Task.WaitAny(new Task[] { testExecution }, _timeout) != -1
-                    || _debugger.IsAttached)
+                var separateContext = new TestExecutionContext(context)
                 {
-                    context.CurrentResult = testExecution.GetAwaiter().GetResult();
-                }
-                else
-                {
-                    string message = $"Test exceeded Timeout value of {_timeout}ms";
+                    CurrentResult = context.CurrentTest.MakeTestResult()
+                };
+                var testExecution = Task.Run(() => innerCommand.Execute(separateContext));
+                var timedOut = Task.WaitAny(new Task[] { testExecution }, _timeout) == -1;
 
+                context.CurrentResult.CopyOutputTo(separateContext.CurrentResult);
+
+                if (timedOut && !_debugger.IsAttached)
+                {
                     context.CurrentResult.SetResult(
                         ResultState.Failure,
-                        message);
+                        $"Test exceeded Timeout value of {_timeout}ms");
+                }
+                else
+                {
+                    context.CurrentResult = testExecution.GetAwaiter().GetResult();
```

Comment:
This is the place where `CurrentResult` gets overwritten and any output created by `SetUp` output would get lost.

Reply:
I moved the `CopyOutputTo` here to make it clear that this is the relevant location.